### PR TITLE
Add unused disk attribute, and fixes clone causing an unused disk.

### LINKED ISSUE #239: When cloning a VM from template...newly created VM got "unused disk 0"
My configuration

Last master branch telmate/proxmox as of 08/11/2020

* Proxmox v 6.2-4

* Terraform v0.13.5
+ provider registry.example.com/telmate/proxmox v2.0.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/null v3.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0

When I am cloning a VM template...newly created VM has got a unused disk...and system want to boot on it....

TIPS :

__Removing this part below from `resource_vm_qemu.go` (line 768) will solve this issue__

```go
err = config.UpdateConfig(vmr, client)
			if err != nil {
				// Set the id because when update config fail the vm is still created
				d.SetId(resourceId(targetNode, "qemu", vmr.VmId()))
				return err
			}
````

At the end go an error though...

```bash
[root@localhost proxmox]# terraform apply -auto-approve 2>&1 | tee tf.log
data.template_file.cloud_init_centos: Refreshing state... [id=9b006253e502423fad4c78fc14ce64c986c284d2ed37184bc23623e436aec5d0]
local_file.cloud_init_centos: Refreshing state... [id=b3896de95e63c8d2c16cf0b1e098110df91efb2a]
null_resource.cloud_init_centos: Refreshing state... [id=7868815670615807967]
proxmox_vm_qemu.vm-test-fred-01: Refreshing state... [id=pve/qemu/102]
proxmox_vm_qemu.vm-test-fred-01: Creating...
proxmox_vm_qemu.vm-test-fred-01: Still creating... [10s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [20s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [30s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [40s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [50s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m0s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m10s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m20s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m30s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m40s elapsed]
proxmox_vm_qemu.vm-test-fred-01: Still creating... [1m50s elapsed]

Error: network.0.firewall: '' expected type 'bool', got unconvertible type 'int'
```

proxmox

Terraform file I am using

```tf
# Create the VM
resource "proxmox_vm_qemu" "vm-test-fred-01" {
  ## Wait for the cloud-config file to exist

  depends_on = [
    null_resource.cloud_init_centos
  ]

  name            = "vm-test-fred-01"
  target_node     = "pve"

  # Clone from debian-cloudinit template
  clone = "CentOS7-Template"
  full_clone      = true
  #os_type = "cloud-init"
  #os_type = "linux"

  # Cloud init options
  cicustom = "user=local:snippets/cloud_init_centos.yml"

  memory       = 512
  agent        = 1

  # Set the boot disk paramters
  bootdisk = "scsi0"
  boot = "cdn"
  scsihw       = "virtio-scsi-single"

  disk {
    #id              = "0"
    size            = "10G"
    type            = "scsi"
    storage         = "local-lvm"        
    #storage_type    = "lvm-thin"
    #valid only with virtio disk or virtio-scsi-single controller
    #iothread        = true 
  }

  # Set the network
  network {
    #id = 0
    model = "virtio"
    bridge = "vmbr0"
  }

  # Cloud Init Settings
  ipconfig0 = "ip=dhcp"

  # Ignore changes to the network
  ## MAC address is generated on every apply, causing
  ## TF to think this needs to be rebuilt on every apply
  lifecycle {
     ignore_changes = [
       network
     ]
  }
}
```

### PR DESCRIPTION
**Pull request by doransmestad**
This PR:

- Adds unused disk attributes for qemu vms
- Fixes #239  where a clone will cause an unused disk  
- Adds provider setting to allow dangerous acceptance of unknown attributes (this will allow bypassing the fatal error triggered when the proxmox api changes see #237)
- Adds more tests.

This pull request depends on  https://github.com/Telmate/proxmox-api-go/pull/99,  once that PR is accepted and this PR is updated with the appropriate go.mod updates this will be ready for merge. 

When using a `replace github.com/Telmate/proxmox-api-go => /src/proxmox-api-go` directive in go.mod to confirm functionality, all acceptance tests pass:

```
$ make acctest

Removing bin/
 -> Building
mkdir -p bin
CGO_ENABLED=0 go build  -o bin/terraform-provider-proxmox_v2.0.0 cmd/terraform-provider-proxmox/*
Built terraform-provider-proxmox
# to run only certain tests, run something of the form:  make acctest TESTARGS='-run=TestAccProxmoxVmQemu_DiskSlot'
TF_ACC=1 go test ./proxmox
ok      github.com/Telmate/terraform-provider-proxmox/proxmox   253.887s
```

### CODE REVIEW

**Comment by ryan4yin** [comment]
this works for me

File: proxmox/provider.go [hunk 1]
```diff
@@ -91,6 +92,12 @@ func Provider() *schema.Provider {
 				Optional: true,
 				Default:  300,
 			},
+			"pm_dangerously_ignore_unknown_attributes": {
```
Comment (ggongaware) [inline proxmox/provider.go:95]: Great name :-)

**Comment by mleone87** [comment]
acceptance test ok also for me(had to change smth in config)

File: proxmox/provider.go (hunk 1 above)
Comment (doransmestad) [reply proxmox/provider.go:95]: haha thanks  :-)

**Comment by doransmestad** [comment]
@mleone87  thanks for checking it!   Glad to hear it worked on your end too.  

I've updated this pull request based on the latest master and proxmox-api-go version.  Should be all set to merge!

**Pull request merged**